schema.py

```diff
@@ -266,6 +267,26 @@ def is_valid(self, data):
         else:
             return True
 
+
+    def set_schema_name(self, error_type):
```

Comment:
This should be a "private" method (i.e. prefixed with an underscore `_`).

Also, I might suggest changing this to be `_raise_error(self, exception, message)` which prepends `self._name` to message if it's not `None` then raises `exception(self._name + " " + message)`. Example usage:

```python
message = "missing keys {keys}".format(keys=the_missing_keys)
self._raise_error(SchemaMissingKeyError, message)
```

Reply:
If it's possible to make this method the one stop shop for error raising inside `validate` that would be amazing since it would clean up a log of the code dedicated to message formatting in it.

Reply:
Big +1 on this
This is ultimately what I'm hoping for and what I originally attempted until I realized accessing the various variables that are unique to each schema error slightly complicates things. Not terribly, but enough that I didn't wanna try to implement anything too drastic right off the bat while I was still getting a feel for the Schema module. But ya, I definitely agree

Reply:
Agreed as well.

Reply:
`_raise_error` method is currently implemented with the signature `(self, exception, message, error_list)` and generates the error message via `"%r %s" % (self._name, message,)` as well as raises the various errors within the function.

The function is always executed with a message so there's no need to have it default to None.